ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/ui/graphics/ros2/RDXROS2HeightMapVisualizer.java

```diff
@@ -102,15 +114,35 @@ public void acceptHeightMapMessage(HeightMapMessage heightMapMessage)
       }
    }
 
-   private void updateGridMapGraphic(HeightMapMessage heightMapMessage)
+   public void acceptGlobalMapMessage(GlobalMapMessage globalMapMessage)
    {
-      executorService.clearQueueAndExecute(() ->
+      if (enableGlobalHeightMapVisualizer.get())
       {
-         if (enableHeightMapVisualizer.get())
+         for (int i = 0; i < globalMapMessage.getGlobalMap().size(); i++)
          {
-            heightMapGraphicNew.generateMeshesAsync(heightMapMessage);
+            acceptGlobalMapTileMessage(globalMapMessage.getGlobalMap().get(i));
          }
-      });
+      }
+   }
+
+   public void acceptGlobalMapTileMessage(GlobalMapTileMessage globalMapTileMessage)
+   {
+      if (enableGlobalHeightMapVisualizer.get())
```

Comment:
Is this if statement checked in the method above? We may be able to remove this and get away with it?

Reply:
Oh I see this is being used in a couple places, this may be fine then